operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -251,6 +255,59 @@ func (r *RedpandaReconciler) reconcileDefluxed(ctx context.Context, rp *v1alpha2
 		return nil
 	}
 
+	// DeepCopy values to prevent any accidental mutations that may occur
+	// within the chart itself.
+	values := rp.Spec.ClusterSpec.DeepCopy()
+
+	objs, err := redpanda.Chart.Render(kube.Config{}, helmette.Release{
+		Namespace: rp.Namespace,
+		Name:      rp.GetHelmReleaseName(),
+		Service:   "Helm",
+	}, values)
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range objs {
+		// Namespace is inconsistently set across all our charts. Set it
+		// explicitly here to be safe.
+		obj.SetNamespace(rp.Namespace)
+		obj.SetOwnerReferences([]metav1.OwnerReference{rp.OwnerShipRefObj()})
+
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+
+		annos := obj.GetAnnotations()
+		if annos == nil {
+			annos = map[string]string{}
+		}
+
+		// Needed for interop with flux.
+		// Without these the flux controller will refuse to take ownership.
+		annos["meta.helm.sh/release-name"] = rp.GetHelmReleaseName()
+		annos["meta.helm.sh/release-namespace"] = rp.Namespace
+
+		labels["helm.toolkit.fluxcd.io/name"] = rp.GetHelmReleaseName()
+		labels["helm.toolkit.fluxcd.io/namespace"] = rp.Namespace
+
+		obj.SetLabels(labels)
+		obj.SetAnnotations(annos)
+
+		if _, ok := annos["helm.sh/hook"]; ok {
+			log.Info(fmt.Sprintf("skipping helm hook %T: %q", obj, obj.GetName()))
+			continue
+		}
+
+		// TODO: how to handle immutable issues?
+		if err := r.apply(ctx, obj); err != nil {
+			return errors.Wrapf(err, "deploying %T %#v %q", obj, obj.GetObjectKind().GroupVersionKind(), obj.GetName())
+		}
+
+		log.Info(fmt.Sprintf("deployed %T: %q", obj, obj.GetName()))
```

Comment:
NIT: There too many resources that have the same name. It would be good to add GroupVersionKind.
```suggestion
		log.Info(fmt.Sprintf("deployed %T %#v %q", obj, obj.GetObjectKind().GroupVersionKind(), obj.GetName()))
```

Reply:
`%T` _kinda_ logs out the GVK: `deployed *v1.Statefulset: "redpanda"`. Would be nice to make it follow the standard kubernetes convention though.